cni/network/plugin/main.go

```diff
@@ -174,6 +175,13 @@ func main() {
 		&network.Multitenancy{},
 		&acnnetwork.AzureHNSEndpoint{},
 	)
+
+	defer func() {
+		if err != nil {
+			os.Exit(1)
+		}
+	}()
```

Comment:
i don't like this defer pattern. it would be better to make a function which can return err, have main only call it and exit with 0 or 1 based on the err, and move all of this in to that function.
short of that...defers go in to a call stack (LIFO), so this would exit before the log.Close above, so this should be the first defer